packages/dds/tree/src/simple-tree/schemaTypes.ts

```diff
@@ -48,6 +48,48 @@ export function isTreeNodeSchemaClass<
  */
 export type AllowedTypes = readonly LazyItem<TreeNodeSchema>[];
 
+/**
+ * Additional information to provide to a Node Schema.
+ *
+ * @typeParam TCustomMetadata - Custom metadata properties to associate with the Node Schema.
+ * See {@link NodeSchemaMetadata.custom}.
+ *
+ * @public
+ */
+export interface NodeSchemaProps<out TCustomMetadata = unknown> {
```

Comment:
As this is only really used as a type constraint for input parameters, there is a different design pattern we can take around strong typing. Rather than requiring inferring type parameters and extending a parameterized version of this type, you could remove the type parameters, then infer those types from the members.

Ex:
```
export interface NodeSchemaOptions {
readonly metadata?: NodeSchemaMetadata;
readonly optionAddedInTheFutureSomeday: Foo;
}
```
Then the schema factory APIs can use these like `TOptions["metadata"]` or `TOptions["optionAddedInTheFutureSomeday"]`

This avoids the order dependent annoyances of getting lists of generic parameter options, and results in much simpler types (this type gets to be non-generic, and when its used, there is a single type parameter constrained to it, no matter no mater many options it captures)